src/Processors/Merges/Algorithms/ReplacingSortedAlgorithm.cpp

```diff
@@ -81,6 +101,13 @@ IMergingAlgorithm::Status ReplacingSortedAlgorithm::merge()
         {
             max_pos = current_pos;
             setRowRef(selected_row, current);
+            if (cleanup)
+                if (assert_cast<const ColumnUInt8 &>(*(*selected_row.all_columns)[is_deleted_column_number]).getData()[selected_row.row_num])
```

Comment:
Could you re-use the assignment above here? This cast isn't cost-free AFAIK, and this is a pretty performance-sensitive part of the code.